### PR TITLE
Hellopush agdroid 605

### PR DESCRIPTION
**Pull request by damienomurchu**
**Issue (JIRA):**

Part of https://issues.jboss.org/browse/AGDROID-605 (Update AeroGear push demos to use notification badges)

**Changes:**

- adds notification badge functionality to hello-push app
- updates gradle build files to target Android-O
- updates aerogear lib versions
- removes jack

**How to test:**

- add google-services.json file to app/
- update push-config.json with your push server credentials
- open app in Android-O, send notification to device and confirm:
  - notification received by device
  - push notification received when app in background
  - notification badge added when app is in background to app icon to indicate unread notification 

@danielpassos @secondsun

### CODE REVIEW

File: HelloPush/app/build.gradle [hunk 1]
```diff
@@ -34,11 +35,11 @@ android {
...
 
 dependencies {
-    compile 'com.android.support:appcompat-v7:25.0.0'
+    compile 'com.android.support:appcompat-v7:26.0.0'
     compile 'org.jboss.aerogear:aerogear-android-core:3.0.0'
     compile 'org.jboss.aerogear:aerogear-android-pipe:3.0.0'
-    compile 'org.jboss.aerogear:aerogear-android-push:4.0.1'
-    compile 'com.google.firebase:firebase-messaging:9.8.0'
+    compile 'org.jboss.aerogear:aerogear-android-push:4.2.0'
+    compile 'com.google.firebase:firebase-messaging:11.0.1'
```
Comment (danielpassos) [inline HelloPush/app/build.gradle]: This is something I'm not sure yet. It needs to come from the push library :/

File: HelloPush/app/build.gradle [hunk 2]
```diff
@@ -34,11 +31,11 @@ android {
...
-    compile 'com.android.support:appcompat-v7:25.0.0'
-    compile 'org.jboss.aerogear:aerogear-android-core:3.0.0'
-    compile 'org.jboss.aerogear:aerogear-android-pipe:3.0.0'
-    compile 'org.jboss.aerogear:aerogear-android-push:4.0.1'
-    compile 'com.google.firebase:firebase-messaging:9.8.0'
+    compile 'com.android.support:appcompat-v7:26.0.1'
+    compile 'org.jboss.aerogear:aerogear-android-core:3.1.0'
+    compile 'org.jboss.aerogear:aerogear-android-pipe:3.2.0'
+    compile 'org.jboss.aerogear:aerogear-android-push:4.2.0'
+    compile 'com.google.firebase:firebase-messaging:11.0.4'
```
Comment (danielpassos) [inline HelloPush/app/build.gradle]: This is something I'm not sure yet. It needs to come from the push library :/

Comment (damienomurchu) [inline HelloPush/app/build.gradle]: just `compile 'org.jboss.aerogear:aerogear-android-push:4.2.0'` I presume? Know that needs to be merged first - ok to bump the versions of core, pipe & firebase ?

Comment (danielpassos) [inline HelloPush/app/build.gradle]: In theory, we should not have core, pipe and firebase-messaging declared on the app, it is a dependency on push and should be automatically downloaded

File: HelloPush/app/src/main/java/org/jboss/aerogear/android/cookbook/hellopush/handler/NotificationBarMessageHandler.java [hunk 3]
```diff
@@ -49,23 +65,54 @@ public void onMessage(Context context, Bundle bundle) {
...
     private void notify(Context context, Bundle bundle) {
         NotificationManager mNotificationManager = (NotificationManager)
                 context.getSystemService(Context.NOTIFICATION_SERVICE);
 
         String message = bundle.getString(UnifiedPushMessage.ALERT_KEY);
         String pushMessageId = bundle.getString(UnifiedPushMessage.PUSH_MESSAGE_ID);
 
+        @SuppressLint("WrongConstant")
         Intent intent = new Intent(context, MessagesActivity.class)
                 .addFlags(PendingIntent.FLAG_UPDATE_CURRENT)
```
Comment (damienomurchu) [inline HelloPush/app/src/main/java/org/jboss/aerogear/android/cookbook/hellopush/handler/NotificationBarMessageHandler.java]: @secondsun Android Studio doesn't seem to like `PendingIntent.FLAG_UPDATE_CURRENT` here - might get your thoughts, or @danielpassos thoughts on it, if you open the code at some point

File: HelloPush/app/build.gradle (hunk 2 above)
Comment (damienomurchu) [inline HelloPush/app/build.gradle]: Got ya.. any reason it's architected the way it is currently? A single dependency on push would be nice, in an ideal world.. :)

Comment (danielpassos) [inline HelloPush/app/build.gradle]: Yes, but it's a long history, it was a but when we were using maven to release our libraries and the aar did not download the dependency itself.

File: HelloPush/app/src/main/java/org/jboss/aerogear/android/cookbook/hellopush/handler/NotificationBarMessageHandler.java (hunk 3 above)
Comment (danielpassos) [inline HelloPush/app/src/main/java/org/jboss/aerogear/android/cookbook/hellopush/handler/NotificationBarMessageHandler.java]: I will take a look soon-ish

Comment (damienomurchu) [inline HelloPush/app/src/main/java/org/jboss/aerogear/android/cookbook/hellopush/handler/NotificationBarMessageHandler.java]: thanks & no rush, not blocked or anything :)

File: HelloPush/app/build.gradle (hunk 2 above)
Comment (damienomurchu) [inline HelloPush/app/build.gradle]: Do you want me to make any changes to the PR re: the above?

Comment (danielpassos) [inline HelloPush/app/build.gradle]: Nah, let's leave it like it

**Comment by danielpassos** [comment]
@damienomurchu @secondsun Just wondering if we should merge it before release the AeroGear Android Push 4.2.0. Wdyt?

**Comment by damienomurchu** [comment]
@danielpassos I think so, seeing as we're just merging it to an android-o branch and not master

**Comment by danielpassos** [comment]
Landed at Android-O branch https://github.com/aerogear/aerogear-android-cookbook/commit/83b08f07329046a379b681698c45402069848601

**Pull request merged**